### PR TITLE
Added openai python

### PR DESCRIPTION
**Pull request by davesarmoury**
Please add the following dependency to the rosdep database.

## Package name:

openai

## Package Upstream Source:

https://github.com/openai/openai-python

## Purpose of using this:

This dependency allows access to OpenAI's servers for off-robot AI processing

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - pip
    - https://pypi.org/project/openai/
- Ubuntu: https://packages.ubuntu.com/
  - pip
    - https://pypi.org/project/openai/
- Fedora: https://packages.fedoraproject.org/
  - pip
    - https://pypi.org/project/openai/

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -295,6 +295,16 @@ opcua-pip:
   ubuntu:
     pip:
       packages: [opcua]
+openai:
```
Comment (clalancette) [inline rosdep/python.yaml:298]: The name of this key should be:

```suggestion
python3-openai-pip:
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by methylDragon — APPROVED** [review]

**Review by tfoote — APPROVED** [review]

File: rosdep/python.yaml [hunk 2]
```diff
@@ -295,6 +295,16 @@ opcua-pip:
   ubuntu:
     pip:
       packages: [opcua]
+python3-openai-pip:
```
Comment (tfoote) [inline rosdep/python.yaml:298]: With the rename it's out of alphabetical order.

**Review by tfoote — CHANGES_REQUESTED** [review]

**Comment by clalancette** [comment]
Totally unrelated to the contents of this PR, this needs a rebase.  If you could rebase this onto the latest, then we can get CI to pass.  Sorry for the inconvenience.

**Review by clalancette — APPROVED** [review]

**Comment by davesarmoury** [comment]
@clalancette  Rebased

File: rosdep/python.yaml [hunk 3]
```diff
@@ -3132,6 +3132,16 @@ python-open3d-pip:
   ubuntu:
     pip:
       packages: [open3d-python]
+python3-openai-pip:
```
Comment (clalancette) [inline rosdep/python.yaml:3135]: Oh, right.  This key also needs to move to be among the `python3-` section, otherwise it is out-of-order alphabetically.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]

**Pull request merged**